source/blog/2021-05-04-announcing-hanami-200alpha2.html.markdown

```diff
@@ -0,0 +1,376 @@
+
+A view in Hanami 2.0 is a standalone, callable class that can declare dependencies to interact with the rest of the application (are you catching the theme here?). It can access parameters and then prepare named exposures to make available to its corresponding template.
+
+```ruby
+module Admin
+  module Views
+    module Articles
+      class Show < Admin::View
+        include Deps["article_repo"]
+
+        expose :article do |id:|
+          article_repo.find(id)
+        end
+      end
+    end
+  end
+end
+```
+
+Every exposure’s value is decorated by a matching view part class, which you can use to provide view-specific behaviour attached to specific domain objects, including anything possible from within the templates, such as rendering partials and accessing all aspects of the general view rendering context.
+
+```ruby
+module Admin
+  module View
+    module Parts
+      class Article < Admin::Part
+        def preview_text
+          body_text.to_s[0..300]
+        end
+
+        def render_social_preview
+          render(:social_preview, title: title, text: preview_text)
+        end
+      end
+    end
+  end
+end
+```
+
+Views also integrate nicely with actions, allowing you to keep your actions clean and focused on HTTP responsibilities only.
+
+```ruby
+module Admin
+  module Actions
+    module Articles
+      class Show < Admin::Action
+        include Deps[view: "views.articles.show"]
+
+        def handle(req, res)
+          res.render view, id: req.params[:id]
+        end
+      end
+    end
+  end
+end
+```
+
+Since views are independent, addressable, callable objects just like any other component within an Hanami application, they can also be put to a wide range of uses alongside the standard rendering of web page HTML, such as rendering emails or even preparing API responses.
+
+## Zero-boilerplate integration of application components
+
+A strong focus of our effort in building Hanami 2.0 has been to allow each component, such as a view or action, to remain useful outside of Hanami, while also fitting seamlessly when used within a full Hanami application. A view used outside of Hanami, for example, looks like this:
+
+```ruby
+class MyView < Hanami::View
+  config.template = "my_view"
+end
+```
+
+And this same view used within Hanami looks like this:
+
+```ruby
+class MyView < Hanami::View
+end
+```
+
+There’s almost no difference! Once you understand how to use `Hanami::View` once, you can use it everywhere. Even inside an Hanami app, where the app seamlessly integrates the views (in the case above, inferring the template name automatically, among other things), you can still access the full extent of the view configuration, allowing you to ”eject” from the configured defaults if you ever need.
```

Comment:
```suggestion
There’s almost no difference! Once you understand how to use `Hanami::View`, you can use it everywhere. Even inside an Hanami app, where the app seamlessly integrates the views (in the case above, inferring the template name automatically, among other things), you can still access the full extent of the view configuration, allowing you to ”eject” from the configured defaults if you ever need.
```

Reply:
Tweak slightly to make it read better, thanks @solnic :)